src/test/java/com/yahoo/bullet/record/SimpleBulletRecordTest.java

```diff
@@ -0,0 +1,552 @@
+
+        newRecord.set("newA", record, "a");
+        Assert.assertEquals(newRecord.get("newA"), "foo");
+        Assert.assertEquals(newRecord.toString(), "{newA:foo}");
+
+        newRecord.set("newB", record, "b", "b.1");
+        Assert.assertEquals(newRecord.get("newB"), "bar");
+
+        newRecord.set("newC", record, "c", 0);
+        Assert.assertEquals(newRecord.get("newC"), singletonMap("c.1", false));
+
+        // The Map isn't copied
+        ((Map<String, Boolean>) record.get("c", 0)).put("c.1", true);
+        Assert.assertEquals(newRecord.get("newC"), singletonMap("c.1", true));
+
+        // And not in the first record either
+        data.put("c.2", false);
+        Assert.assertEquals(((Map) record.get("c", 0)).get("c.2"), false);
+        Assert.assertEquals(newRecord.get("newC", "c.2"), false);
+    }
+
+    @Test
+    public void testGetTopLevelField() {
+        SimpleBulletRecord record = new SimpleBulletRecord();
+        record.setListOfStringMap("foo", singletonList(singletonMap("qux", "norf")));
+        record.setString("bar", null);
+
+        Assert.assertNull(record.get("bar"));
+        Assert.assertTrue(record.get("foo") instanceof List);
+        Assert.assertEquals(record.get("foo"), singletonList(singletonMap("qux", "norf")));
+        Assert.assertNull(record.get("dne"));
+    }
+
+    @Test
+    public void testGetMapField() {
+        SimpleBulletRecord record = new SimpleBulletRecord();
+        record.setMap("foo", Pair.of("bar", "baz"), Pair.of("baz", "qux"));
+
+        Assert.assertTrue(record.get("foo") instanceof Map);
+        Assert.assertEquals(record.get("foo", "bar"), "baz");
+        Assert.assertEquals(record.get("foo", "baz"), "qux");
+
+        Assert.assertNull(record.get("dne"));
+        Assert.assertNull(record.get("foo", null));
+        Assert.assertNull(record.get("foo", ""));
+        Assert.assertNull(record.get("foo", "dne"));
+    }
+
+    @Test
+    public void testGetListIndex() {
+        SimpleBulletRecord record = new SimpleBulletRecord();
+        record.setListMap("foo", singletonMap("a", 1L), singletonMap("b", 2L));
+        Assert.assertTrue(record.get("foo") instanceof List);
+        Assert.assertEquals(record.get("foo", 0), singletonMap("a", 1L));
+        Assert.assertEquals(record.get("foo", 1), singletonMap("b", 2L));
+
+        Assert.assertNull(record.get("dne", -1));
+        Assert.assertNull(record.get("dne", 0));
+        Assert.assertNull(record.get("dne", 123123));
+    }
+
+    @Test(expectedExceptions = IndexOutOfBoundsException.class)
+    public void testGetListBadIndex() {
+        SimpleBulletRecord record = new SimpleBulletRecord();
+        record.setListMap("foo", singletonMap("a", 1L));
+        record.get("foo", 1);
+    }
+
+    @Test
+    public void testIterator() {
+        SimpleBulletRecord record = new SimpleBulletRecord();
+        record.setMap("4", Pair.of("4.1", false))
+                .setString("1", "bar").setLong("2", 42L)
```

Comment:
Minor Comment:  Our convention has been to line up the indentation with the previous function call in the line above.  It looks like the indentation got changed in this file (probably by an IDE?).  Maybe Fix the indentation in this file (several places all through the file) to match our current convention?

Reply:
Done. It was by IntelliJ.